### PR TITLE
Allow users to disable docker

### PR DESCRIPTION
**Pull request by edulop91**
Partial fix for #174 

Allows users to skip using docker by specifying `FOGG_DISABLE_DOCKER`. There is still work needed to:
* Make sure we install terraform and dependencies where needed
* Deal with custom plugins and providers

Should be no breaking changes in this pr

### CODE REVIEW

File: templates/repo/scripts/Makefile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ifeq ($(shell uname -m),i686)
+  ARCH ?= "386"
+endif
+ifeq ($(shell uname -m),aarch64)
+  ARCH ?= "arm"
+endif
+
+TF_PRESENT := $(shell command -v terraform 2> /dev/null)
+TF_CURRENT_VERSION := $(shell terraform version | head -n1 | sed 's/Terraform v//g')
+TF_VERSION := 0.11.8
```
Comment (ryanking) [inline templates/repo/scripts/Makefile]: this needs to come from a template var

File: templates/repo/scripts/Makefile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+TFLINT_PRESENT := $(shell command -v tflint 2> /dev/null)
```
Comment (ryanking) [inline templates/repo/scripts/Makefile]: we can do this in a separate pr, but tflint should get installed like a plugin

File: templates/repo/scripts/Makefile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+endif
+
+tflint-required:
+ifndef TFLINT_PRESENT
+	$(error "tflint is not installed")
+endif
+
+install: install-terraform install-tflint
+
+install-terraform:
```
Comment (ryanking) [inline templates/repo/scripts/Makefile]: should we consider https://github.com/Zordrak/tfenv ?

File: templates/repo/scripts/Makefile (hunk 1 above)
Comment (edulop91) [reply templates/repo/scripts/Makefile]: Oh yeah, pr still needs a bunch of work. Opened to get eyes on general approach

File: templates/repo/scripts/Makefile (hunk 2 above)
Comment (edulop91) [reply templates/repo/scripts/Makefile]: Yeah, still thinking how to reconcile plugins feature with no docker

Comment (ryanking) [reply templates/repo/scripts/Makefile]: I haven't fully thought this through, but was thinking of managing a `bin` directory in the repo (or one for each arch) and adding that to the `PATH` in the makefiles.

Comment (edulop91) [reply templates/repo/scripts/Makefile]: yeah, i was thinking something similar. will do in the next pr though so this doesn't get too big

File: templates/repo/scripts/component.mk [hunk 4]
```diff
@@ -10,46 +10,57 @@ AUTO_APPROVE := false
...
 	-e RUN_USER_ID=$(shell id -u) -e RUN_GROUP_ID=$(shell id -g) \
 	-e TF_PLUGIN_CACHE_DIR="/repo/.terraform.d/plugin-cache" -e TF="$(TF)" \
 	-w /repo/$(REPO_RELATIVE_PATH) $(TF_VARS) $(FOGG_DOCKER_FLAGS) $$(sh $(REPO_ROOT)/scripts/docker-ssh-mount.sh)
-docker_terraform = $(docker_base) chanzuckerberg/terraform:$(IMAGE_VERSION)
-docker_sh = $(docker_base) --entrypoint='/bin/sh' chanzuckerberg/terraform:$(IMAGE_VERSION)
+docker_terraform := $(docker_base) chanzuckerberg/terraform:$(IMAGE_VERSION)
+docker_sh := $(docker_base) --entrypoint='/bin/sh' chanzuckerberg/terraform:$(IMAGE_VERSION)
+
+ifdef FOGG_DISABLE_DOCKER
+	sh_command ?= $(SHELL)
```
Comment (edulop91) [inline templates/repo/scripts/component.mk]: or should we pick a shell?

Comment (ryanking) [reply templates/repo/scripts/component.mk]: this is probably fine for now
